### DISCUSSION TITLE
Easiest way to add thumbnail cleaner?

### DISCUSSION THREAD

**Discussion by airjer** [ubergeek77/Lemmy-Easy-Deploy#100] [Q&A] [answered]
https://github.com/wereii/lemmy-thumbnail-cleaner

What's the easiest way of going about adding this service?

**Comment by ubergeek77** [comment]
If you don't want to manage your own template manually every time I make changes, you could do something like this:

```bash
#!/bin/bash

LIVE_COMPOSE="/path/to/Lemmy-Easy-Deploy/live/docker-compose.yml"
cp "${LIVE_COMPOSE:?}" "${LIVE_COMPOSE:?}.bak"

if ! grep -qe ' cleaner:' "${LIVE_COMPOSE:?}"; then
  echo "---> Lemmy Thumbnail Cleaner not found, adding to deployment"
  awk '/^volumes:/{print NR; exit}' "${LIVE_COMPOSE:?}" | xargs -I {} sed -i '{}i\
  cleaner:\
    image: ghcr.io/wereii/lemmy-thumbnail-cleaner:v0.1.3\
    #restart: unless-stopped\
    environment:\
      - RUST_LOG=info\
      - INSTANCE_HOST=https://your_instance_host.here/\
      - POSTGRES_DSN=postgresql://user:password@postgres/lemmy\
      - PICTRS_HOST=pict-rs:8080\
      - PICTRS_API_KEY=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX\
      #- THUMBNAIL_MIN_AGE_MONTHS=3\
      #- CHECK_INTERVAL=300\
      #- QUERY_LIMIT=100\
  ' "${LIVE_COMPOSE:?}"
else
  echo "---> Lemmy Thumbnail Cleaner was already added to this deployment"
fi
```

Save this to `./custom/pre-deploy.sh` and make sure it's executable.

Please don't forget to change whatever values you need to here, and mind the spacing in `awk` when you save the file. It will literally interpret whatever tabs and spaces are there in that block.

Let me know if that works! I don't think many people use pre-deploy scripts like this.

**Reply by airjer** [reply]
```
--> Running custom pre-deploy script
./custom/pre-deploy.sh: line 4: SCRIPT_DIR: parameter null or not set

```

For some reason it doesn't compute the script directory.

**Reply by ubergeek77** [reply]
Aw, I really thought that would have been inherited:

https://github.com/ubergeek77/Lemmy-Easy-Deploy/blob/9722d9f7c312ee46c3d68feab0361b2f2fefec4b/deploy.sh#L1710

Well, since your directories are unlikely to change frequently, you can get away with manually specifying the path to your docker compose file in this script. I've edited my original answer as an example.

**Reply by airjer** [reply]
```
---> Lemmy Thumbnail Cleaner not found, adding to deployment
sed: -e expression #1, char 125: unknown command: `-'

```

Appreciate the help.

**Reply by ubergeek77** [reply]
Can I see what changes you made? That doesn't seem right 🤔

**Reply by airjer** [reply]
Hey that was my bad as I just copied my docker-compose settings I did when trying to do it myself and left off the trailing `\'s`. 

Getting this error I was getting when trying to do it before asking for help. 

```
--> Running custom pre-deploy script
---> Lemmy Thumbnail Cleaner not found, adding to deployment
yaml: line 81: did not find expected key
```

**Reply by ubergeek77** [reply, marked as answer]
Alright, yaml doesn't like the spacing. Let's make it a little more predictable and readable:

```
#!/bin/bash

mkdir -p ./custom
LIVE_COMPOSE="./live/docker-compose.yml"

(
  echo "  cleaner:"
  echo "    image: ghcr.io/wereii/lemmy-thumbnail-cleaner:v0.1.3"
  echo "    #restart: unless-stopped"
  echo "    environment:"
  echo "      - RUST_LOG=info"
  echo "      - INSTANCE_HOST=https://your_instance_host.here/"
  echo "      - POSTGRES_DSN=postgresql://user:password@postgres/lemmy"
  echo "      - PICTRS_HOST=pictrs:8080"
  echo "      - PICTRS_API_KEY=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
  echo "      #- THUMBNAIL_MIN_AGE_MONTHS=3"
  echo "      #- CHECK_INTERVAL=300"
  echo "      #- QUERY_LIMIT=100"
  echo ""
) > ./custom/lemmy-thumbnail-cleaner.snip

if ! grep -qe ' cleaner:' "${LIVE_COMPOSE:?}"; then
  echo "---> Lemmy Thumbnail Cleaner not found, adding to deployment"
  sed -i '/^volumes:/e cat ./custom/lemmy-thumbnail-cleaner.snip' "${LIVE_COMPOSE:?}"
else
  echo "---> Lemmy Thumbnail Cleaner was already added to this deployment"
fi
```

In theory you shouldn't need to change any paths with this, because the pre deploy script executes from within the LED folder.

**Reply by ubergeek77** [reply]
I've just tested this new script on my own deployment and it seems to be working

Note that I did have an error in a path originally, but I just edited it

**Reply by airjer** [reply]
Nice it's working, I had to take the dash out of pict-rs to get the host name resolving correctly. 

Thank you!
